business-registry-dashboard/app/pages/incorporateNow.vue

```
@@ -72,34 +70,41 @@ onMounted(async () => {
       }
     }
 
-    // Load user's affiliated businesses and check if this name request is already affiliated
-    // This prevents unauthorized access to name requests that belong to other users
+    // Load and check if the NR is already affiliated
     await affStore.loadAffiliations()
     const isAffiliated = isNameRequestAffiliated(affStore.affiliations.results, business.nameRequest?.nrNumber)
 
-    // Verify business eligibility and proceed with incorporation
-    if (business.nameRequest.requestActionCd === NrRequestActionCodes.NEW_BUSINESS &&
-        ELIGIBLE_CORP_TYPES.includes(business.nameRequest.legalType) && isAffiliated) {
-      try {
-        // Create the business and redirect to dashboard
-        const identifier = await createBusinessFromNR(business)
-        await affNav.goToDashboard(identifier)
-      } catch (error) {
-        console.error('Error creating named business:', error)
-        brdModal.openMagicLinkModal(
-          t('error.magicLinkCreateBusinessFailed.title'),
-          t('error.magicLinkCreateBusinessFailed.description')
-        )
-      }
+    // Create affiliation if it doesn't exist
+    if (!isAffiliated) {
+      await affStore.createNRAffiliation({
+        businessIdentifier: business.nameRequest?.nrNumber,
+        ...(phone && { phone }),
+        ...(email && { email })
+      })
+    }
+
+    // Get the temporary business identifier if it exists
+    const tempBusinessIdentifier = getTempBusinessIdentifierOfNameRequest(affStore.affiliations.results, business.nameRequest?.nrNumber)
+
+    // Check basic eligibility conditions
+    const isEligibleRequest = business.nameRequest.requestActionCd === NrRequestActionCodes.NEW_BUSINESS &&
+      ELIGIBLE_CORP_TYPES.includes(business.nameRequest.legalType)
+
+    if (!isEligibleRequest) {
+      throw new Error('Invalid name request')
+    }
```

I recommend moving these checks before the affiliation above.

---

In general, do the checks in a order that will fail fastest for the user.